### PR TITLE
board: arm64: Add suport for ROC-RK3568-PC

### PR DESCRIPTION
**Pull request by cocoeoli**
This is support for AArch64 development board.
The board uses 4-core Cortex-A55, which are based on the ARMv8.2 architecture.
In addition,we support smp support and
it can use 4-cores to run basic samples.
Associated PR: #63343

### CODE REVIEW

File: soc/arm64/rockchip/rk3568/soc.c [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Copyright 2022 openEuler SIG-Zephyr
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+
+#ifdef CONFIG_SOC_RK3568
+
+#ifndef CONFIG_PM_CPU_OPS_PSCI
+int pm_cpu_on(unsigned long cpuid, uintptr_t entry_point)
```
Comment (carlocaione) [inline soc/arm64/rockchip/rk3568/soc.c:12]: why is this needed? If `CONFIG_PM_CPU_OPS_PSCI` is not defined, `pm_cpu_on` should return `-ENOTSUP` (if `CONFIG_PM_CPU_OPS` is defined).

**Review by carlocaione — CHANGES_REQUESTED** [review]

File: soc/arm64/rockchip/rk3568/soc.c (hunk 1 above)
Comment (cocoeoli) [reply soc/arm64/rockchip/rk3568/soc.c:12]: Thanks, it seem not necessary for specific scenarios. I think there may be situations where `CONFIG_PM_CPU_OPS_PSCI` is not defined but `pm_cpu_on` is used.

File: arch/arm64/core/reset.c [hunk 2]
```diff
@@ -125,6 +125,11 @@ void z_arm64_el2_init(void)
 	write_sctlr_el2(reg);
 
 	reg = read_hcr_el2();
+	/* when EL2 is enable in current security status:
+	 * Clear TGE bit: All exceptions that would not be routed to EL2;
+	 * Clear IMO bit: Physical IRQ interrupts are not taken to EL2 and EL3.
```
Comment (carlocaione) [inline arch/arm64/core/reset.c:130]:
```suggestion
	 * Clear TGE bit: All exceptions that would be routed to EL2 are routed to EL1;
	 * Clear IMO bit: Physical IRQ interrupts are not taken to EL2 and EL3.
```

I'm a bit lost on the IMO bit. According to the documentation: `If the value of HCR.TGE is 0, then virtual SError interrupts are enabled in Non-secure state.`. Where did you get that comment regarding physical IRQs?

**Review by carlocaione — CHANGES_REQUESTED** [review]

File: arch/arm64/core/reset.c (hunk 2 above)
Comment (cocoeoli) [reply arch/arm64/core/reset.c:130]: In the Arm® Architecture Reference Manual(page D13-3336), the HCR_EL2 register's IMO bit has below feature: when IMO bit is 1, `Physical IRQ interrupts are taken to EL2, unless they are routed to EL3.`. 
image

What you describe may be the HCR_ELR.AMO bit:
image
So, I clear HCR_EL2.IMO bit.

Comment (povergoing) [reply arch/arm64/core/reset.c:130]: Then do we need also to clear the AMO bit?
Also, it would be better to separate the changes in `reset.c` and `cpu.h` into a new commit.

Comment (cocoeoli) [reply arch/arm64/core/reset.c:130]: Thanks,  the reason why there is no fatal now may be that no SError occurred.  But when zephyr boot using uboot on roc_rk3568_pc, HCR_EL2.AMO bit are set to 1 too.  Clear it may be a good choice. 
On some platforms such as qemu_ cortex_ a53, this operation is not necessary, because the value of HCR_EL2 register is zero. I will add a new commit to annouce it.

**Review by carlocaione — CHANGES_REQUESTED** [review]
nit: Can you fix spelling and spacing in the first commit title?

`arch: arm64: Re-init HCR_EL2 in z_arm64_el2_init`

File: arch/arm64/core/reset.c [hunk 3]
```diff
@@ -125,6 +125,13 @@ void z_arm64_el2_init(void)
 	write_sctlr_el2(reg);
 
 	reg = read_hcr_el2();
+	/* when EL2 is enable in current security status:
+	 * Clear TGE bit: All exceptions that would not be routed to EL2;
+	 * Clear AMO bit: Physical SError interrupts are not taken to EL2 and EL3.
+	 * Clear IMO bit: Physical IRQ interrupts are not taken to EL2 and EL3.
+	 */
+	reg &= ~(HCR_IMO_BIT |
+		HCR_AMO_BIT | HCR_TGE_BIT);
```
[the remark this answers is not in the document]
Comment (cocoeoli) [reply arch/arm64/core/reset.c]: Ok, I have changed the title.

**Review by carlocaione — DISMISSED** [review]

**Comment by povergoing** [comment]
Can you please fix the CI failures?

**Comment by cocoeoli** [comment]
> Can you please fix the CI failures?
Some failure are caused by building other board, such as `west twister -p circuitdojo_feather_nrf9160_ns -s tests/kernel/common/kernel.common` will cause below failures:
image
Could this be considered as an issue with the test of `circuitdojo_feather_nrf9160_ns` board?

**Comment by cocoeoli** [comment]
> Can you please fix the CI failures?

Yes, I modified the documentation, it seems work.

**Comment by cocoeoli** [comment]
> Can you please fix the CI failures?

I am not very familiar with Github CI. Can you tell me why the build process of other board affects this PR's check?
`west build -p -b qemu_xtensa_mmu tests/kernel/fatal/message_capture -T kernel.logging.message_capture`

**Comment by carlocaione** [comment]
@cocoeoli try to rebase on the latest `main`

**Comment by cocoeoli** [comment]
> @cocoeoli try to rebase on the latest `main`

Thanks, I will have a try.

**Comment by cocoeoli** [comment]
@carlocaione @povergoing Please help me review these commits and see if there are any other issues?

**Review by povergoing — APPROVED** [review]
LGTM

**Review by carlocaione — APPROVED** [review]

**Comment by Cherish-Gww** [comment]
@cocoeoli 
Hi, I have run this patch on my RK3566 platform, that cool, next I want try zvm on my platform.
